lib/charms/operator_libs_linux/v0/systemd.py

```diff
@@ -0,0 +1,148 @@
+# Copyright 2021 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""Abstractions for stopping, starting and managing system services via systemd.
+
+This library assumes that your charm is running on a platform that uses systemd. E.g.,
+Centos 7 or later, Ubuntu Xenial (16.04) or later.
+
+For the most part, we transparently provide an interface to a commonly used selection of
+systemd commands, with a few shortcuts baked in. For example, service_pause and
+service_resume with run the mask/unmask and enable/disable invocations.
+
+TODO: do we want to add some magic to service_start and service_restart, so that you can
+run them without checking to see if the service is running first? E.g., service_start
+either starts or restarts a service, and service_restart either restarts or starts the
+service.
+
+Example usage:
+```python
+from systemd import *
+
+# Start a service
+if not service_running("mysql"):
+    success = service_start("mysql")
+
+# Attempt to reload a service, restarting if necessary
+success = service_reload("nginx", restart_on_failure=True)
+
+```
+
+"""
+
+import logging
+import subprocess
+
+__all__ = [  # Don't export `service`. (It's not the intended way of using this lib.)
+    "service_pause",
+    "service_reload",
+    "service_restart",
+    "service_resume",
+    "service_running",
+    "service_start",
+    "service_stop",
+]
+
+logger = logging.getLogger(__name__)
+
+
+def service(action: str, service_name: str) -> bool:
```

Comment:
How about adding an optional list of arguments? This would allow passing `--quiet` for `is-active` or `--now` for `enable/disable`.

Reply:
I wanted to avoid pulling in a lot of code that setup and configured the arg parse lib to handle this. We could do a naive thing, where we just accept an *args that is a list of strings that you want to pass in. That feels a bit hacky, though. 

For now, I'd like to submit the library as is, and then make a separate PR for an enhancement that allows for passing in flags.